### PR TITLE
[14.0] mail_notification_clean_status_error: remove notifications in error

### PR DESCRIPTION
**Pull request by sebalix**
Delete email notifications in error.

The scheduled action *Notification: Delete Notifications older than 6 Month* provided by Odoo is responsible to delete notifications that have been sent successfully.

However, it doesn't delete the notifications that could not be sent, and their number could keep growing over time, impacting the performance of some queries related to the chatter.

This module extends the scheduled action of Odoo to also delete such notifications.

On a DB with 20M+ of `mail_message` records and 100k notifications in error over 1M+, this allows to speed up the loading of the Discuss app widget (top bar) and the chatter of records from ~35s to ~1s.

Ref. 4352

### CODE REVIEW

**Review by hbrunn — APPROVED** [review]
I'd like to have a test for this though

File: mail_notification_clean_status_error/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+{
+    "name": "Mail Notification - Clean up notifications in error",
+    "summary": "Extend Odoo scheduled task to also delete notifications in error.",
+    "version": "14.0.1.0.0",
+    "category": "Social Network",
```
Comment (simahawk) [inline mail_notification_clean_status_error/__manifest__.py]:
```suggestion
    "category": "Mail",
```

**Review by simahawk — COMMENTED** [review]
Tests required :wink:

**Comment by sebalix** [comment]
@hbrunn @simahawk test added

**Review by simahawk — APPROVED** [review]

**Pull request merged**